### PR TITLE
feat: allow doctrine/orm 3, doctrine/dbal 4 (WIP)

### PR DESCRIPTION
**Pull request by Chris53897**
jackalope/jackalope-doctrine-dbal does not yet support dbal 4 (but they are working on it)

### CODE REVIEW

**Comment by Chris53897** [comment]
Now an installable set is available.

There are multiple errors.
One is this, that change was intoduced in doctrine/orm 2.16.x (later removed) and reintoduced in 3.x.
If someone knows how to fix it, please let me know.

 ```
Doctrine\ORM\Exception\EntityIdentityCollisionException: While adding an entity of class Liip\Acme\Tests\App\Entity\User with an ID hash of "1" to the identity map,
   │ another object of class Liip\Acme\Tests\App\Entity\User was already present for the same ID. This exception
   │ is a safeguard against an internal inconsistency - IDs should uniquely map to
   │ entity object instances. This problem may occur if:
   │
   │ - you use application-provided IDs and reuse ID values;
   │ - database-provided IDs are reassigned after truncating the database without
   │ clearing the EntityManager;
   │ - you might have been using EntityManager#getReference() to create a reference
   │ for a nonexistent ID that was subsequently (by the RDBMS) assigned to another
   │ entity.
   │
   │ Otherwise, it might be an ORM-internal inconsistency, please report it.
```

**Comment by mbabker** [comment]
> One is this, that change was intoduced in doctrine/orm 2.16.x (later removed) and reintoduced in 3.x.
> If someone knows how to fix it, please let me know.

Just looking at the test fixtures, I don't have a good solution here.  But, all the `setId()` calls are probably what is causing the ORM's identity map exception.  If the tests can be updated in a way that they don't rely on specific primary key values then that would be the best course of action.

File: tests/App/config.yml [hunk 1]
```diff
@@ -29,15 +29,10 @@ doctrine:
                 connection: default
                 mappings:
                     LiipAcme:
-                        type: php
+                        type: attribute
```
Comment (mbabker) [inline tests/App/config.yml]: You'll need support for annotation mapping here as well, otherwise you'll need to skip tests on PHP 7.4 builds.

File: tests/App/config.yml [hunk 2]
```diff
@@ -29,15 +29,10 @@ doctrine:
...
                         dir: "%kernel.project_dir%/Entity"
                         prefix: 'Liip\Acme\Tests\App\Entity'
                         is_bundle: false
-                    LiipAcmeYml:
```
Comment (mbabker) [inline tests/App/config.yml]: If the intent is to test with multiple mapping drivers, converting to XML instead of dropping the YAML mapping would be a good choice (you should be able to use the CLI tools when ORM 2.x is installed to help with this).

Comment (Chris53897) [reply tests/App/config.yml]: I got the message, that the yamlDriver ist not available anymore.
Good point with the XMLDriver as replacement.

**Comment by Chris53897** [comment]
@mbabker Thanks for the input.

My primary goal is to test my bundle via this bundle ;)

File: composer.json [hunk 3]
```diff
@@ -1,78 +1,80 @@
 {
-    "name": "liip/test-fixtures-bundle",
```
Comment (alexisLefebvre) [inline composer.json:2]: Please restore the original indentation or do this change in another PR.

File: tests/App/config.yml (hunk 2 above)
Comment (alexisLefebvre) [reply tests/App/config.yml]: I don't remember why tests used annotations **and** YAML config, we can drop it and just assume that Doctrine read the configuration properly.

**Comment by Chris53897** [comment]
I will come back to this as soon, as i have some free time.

**Comment by Chris53897** [comment]
I did split the PRs https://github.com/liip/LiipTestFixturesBundle/pull/269
and https://github.com/liip/LiipTestFixturesBundle/pull/268

After that is merged, maybe just simpify things and https://github.com/liip/LiipTestFixturesBundle/issues/272 and add support for doctrine/orm 3, doctrine/dbal 4 after that?

**Comment by alexislefebvre** [comment]
You can rebase it on `2.x`.

Or apply these changes from the `3.x` branch and target this branch when creating another PR, see: 

- #272

**Comment by alexislefebvre** [comment]
I think it has been partially fixed by this PR:

- #296

`doctrine/dbal` 4 is not supported yet.

**Comment by Chris53897** [comment]
thanks @alexislefebvre
I did rebase against 2.x.
There are no useful changes in this PR anymore.
I will close this PR.
